File: models/message.go
package models

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type Message struct {
	ID             uuid.UUID       `db:"id" json:"id"`
	ConversationID uuid.UUID       `db:"conversation_id" json:"conversation_id"`
	Seq            int64           `db:"seq" json:"seq"`
	FromUID        uuid.UUID       `db:"from_uid" json:"from_uid"`     // UID string
	Body           json.RawMessage `db:"body" json:"body"`             // flexible schema for now
	CreatedAt      time.Time       `db:"created_at" json:"created_at"` // ***** type incorrect should reveive number
}

File: web/src/wire/message.ts
import type { Message } from "../context/ChatContext";

export type MsgDto = {
  id: string;
  conversation_id: string;
  seq: number;
  from_uid: string;
  created_at: number;
  body: {
    text: string;
  };
};

export const dtoToMessage = (w: MsgDto): Message => ({
  id: w.id,
  convId: w.conversation_id,
  seq: w.seq,
  fromUid: w.from_uid,
  ts: w.created_at,
  text: w.body?.text ?? "", // safely get text from body
});

export const jsonToMessage = (json: any): Message => ({
  id: json.server_msg_id,
  convId: json.conversation_id,
  seq: json.seq,
  fromUid: json.from_uid,
  ts: json.ts,
  text: json.body?.text ?? "",
});